search/quality_eval.go

```diff
@@ -698,7 +703,8 @@ func EvaluateQuery(q EvalQuery, serverUrl string, skipExpectations bool) EvalRes
 					}
 					if HitMatchesExpectation(hit, hitSource, q.Expectations[i]) {
 						rank = j + 1
-						if j <= 2 {
+						//set good if first 3 are till 3 place in results, or 4 and 5 are till 5 place
+						if j <= 2 || (i > 2 && j <= 4) {
```

Comment:
So you actually constrain good up to 5th place only if expectations index > 2

Should this be
j <= math.Max(len(q.Expectations), 2)

Basically when we have >2 expectations we still consider good when result at <= len(expectations)

Which is can be improved by checking that all the results above i are expected (and not some garbage...)

Give it a though... but seems okay.

Reply:
Your logic does not consider the expectation index... if our result is at 4th place (j=3) and the expectation index is 3 (i=3) then we don't want to mark it as good.
But `j <= math.Max(len(q.Expectations), 2)` will be evaluated as 'true'.